Smell documentation link

danielmbarlow opened the pull request:
Added links on smells to Reek documentation on Github in the code screen. All comments are welcome. We don't use Minitest at work, so let me know if there's anything you want me to change with the tests.

troessner commented on lib/rubycritic/core/smell.rb:
```diff
@@ -46,6 +46,14 @@ def to_json(*a)
       to_h.to_json(*a)
     end
 
+    def dasherized_type
```
Lets make this method private, it doesnt have to be public.

troessner commented on test/lib/rubycritic/core/smell_test.rb:
```diff
@@ -69,4 +69,27 @@
     end
   end
+
+  describe '#dasherized_type' do
```
I believe we should test this method implicitly through doc_url  - how about moving those tests there?

troessner commented:
Btw. this functionality (obviously) exists in Reek as well, unfortunately it's not really well exposed:

```Ruby
warning = FactoryGirl.build(:smell_warning)
Reek::Report::WikiLinkWarningFormatter.new.send :explanatory_link, warning

>>
=> "https://github.com/troessner/reek/blob/master/docs/Feature-Envy.md"

It probably would make sense to expose this in a better way. But that would require a little more sophisticated approach than just making the method public ;)

I guess for now it's better to go with your solution.
Two small comments and the lgtm

danielmbarlow commented:
Hi Timo, I made the changes you suggested, squashed my commits (as I think your CONTRIBUTE.md suggests), and rebased on master. Let me know if you need anything else done for this PR.

troessner commented:
Looks good now!  But travis rightfully complains about rubocop violations ;)

danielmbarlow commented:
Hi Timo, how are we looking now?

troessner commented:
Looks good now! travis is still red because of jruby but that shouldn't be an issue after https://github.com/whitesmith/rubycritic/pull/128 -> merged!

troessner commented:
Btw. extremely awesome pull request, a small change but one that increases convenience significantly, thanks for that!

danielmbarlow commented:
You are welcome. I look forwards to using it. Thanks for Rubycritic!

Onumis commented:
This is so awesome! Thanks @danielmbarlow
